### PR TITLE
Add PHP 8 compatiblity; end PHP 7.2 support

### PR DESCRIPTION
**Pull request by janlam7**
- Adds PHP 8 compatiblity
- Ends PHP 7.2 support

### CODE REVIEW

**Review by svandervlugt — APPROVED** [review]

File: src/Reflection/ReflectionClass.php [hunk 1]
```diff
@@ -88,6 +88,11 @@ class ReflectionClass
      */
     public function __construct($filename)
     {
+        // PHP 8.0 forward compatibility @see https://www.php.net/manual/en/migration80.incompatible.php
+        if (!defined('T_NAME_QUALIFIED')) {
```
Comment (janlam7) [inline src/Reflection/ReflectionClass.php:92]: The TokenStream class has similar but different problems

Comment (janlam7) [reply src/Reflection/ReflectionClass.php:92]: Skipped the TokenStreamTest for PHP8 for now.

**Review by nutama — APPROVED** [review]

File: test/Reflection/TokenStreamTest.php [hunk 2]
```diff
@@ -28,12 +29,24 @@ protected function setUp(): void
...
     public function typeProvider(): array
     {
+        if (version_compare(PHP_VERSION, '8.0.0') >= 0) {
+            return [
+                [         0, T_OPEN_TAG                                ],
+                [         1, T_NAMESPACE                               ],
+                [        4, ';'                                       ],
+                [        -1, null,         \OutOfBoundsException::class],
+                [self::PHP_8_SIZE, null,         \OutOfBoundsException::class],
+                [        32, T_PRIVATE                                 ],
```
Comment (yannickl88) [inline test/Reflection/TokenStreamTest.php]: this feels like it was meant to align on the `,` and the last `]`. Maybe consider aligning it or remove all the unneeded whitespaces if we would align it "regularly"

File: test/Reflection/TokenStreamTest.php [hunk 3]
```diff
@@ -28,12 +29,24 @@ protected function setUp(): void
...
+        }
+
+        // before php7
         return [
             [         0, T_OPEN_TAG                                ],
             [         1, T_NAMESPACE                               ],
             [        10, ';'                                       ],
             [        -1, null,         \OutOfBoundsException::class],
-            [self::SIZE, null,         \OutOfBoundsException::class],
+            [self::PHP_7_SIZE, null,         \OutOfBoundsException::class],
```
Comment (yannickl88) [inline test/Reflection/TokenStreamTest.php]: same here and following cases

File: test/Reflection/ReflectionClassTest.php [hunk 4]
```diff
@@ -147,7 +147,6 @@ public function fileProvider(): array
                     'private $color = COLOR;',
                     'private $color = self::COLOR;',
                     'private $color = An\Other\Place::class;',
-                    'private $color = \An\Other\Place::class;',
```
Comment (yannickl88) [inline test/Reflection/ReflectionClassTest.php]: why was this removed? Do we no longer support it?

**Review by yannickl88 — APPROVED** [review]
codestyle changes are not blocking imo, can be solved later. So approved!
